cmd/soroban-rpc/internal/methods/get_transactions.go

```diff
@@ -0,0 +1,230 @@
+package methods
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"io"
+
+	"github.com/creachadair/jrpc2"
+	"github.com/creachadair/jrpc2/handler"
+	"github.com/stellar/go/ingest"
+	"github.com/stellar/go/support/errors"
+	"github.com/stellar/go/support/log"
+	"github.com/stellar/go/toid"
+
+	"github.com/stellar/soroban-rpc/cmd/soroban-rpc/internal/db"
+	"github.com/stellar/soroban-rpc/cmd/soroban-rpc/internal/ledgerbucketwindow"
+)
+
+// TransactionsPaginationOptions defines the available options for paginating through transactions.
+type TransactionsPaginationOptions struct {
+	Cursor *toid.ID `json:"cursor,omitempty"`
+	Limit  uint     `json:"limit,omitempty"`
+}
+
+// GetTransactionsRequest represents the request parameters for fetching transactions within a range of ledgers.
+type GetTransactionsRequest struct {
+	StartLedger uint32                         `json:"startLedger"`
+	Pagination  *TransactionsPaginationOptions `json:"pagination,omitempty"`
+}
+
+// isValid checks the validity of the request parameters.
+func (req GetTransactionsRequest) isValid(maxLimit uint, ledgerRange ledgerbucketwindow.LedgerRange) error {
+	if req.Pagination != nil && req.Pagination.Cursor != nil {
+		if req.StartLedger != 0 {
+			return errors.New("startLedger and cursor cannot both be set")
+		}
+	} else if req.StartLedger < ledgerRange.FirstLedger.Sequence || req.StartLedger > ledgerRange.LastLedger.Sequence {
+		return errors.Errorf("start ledger must be between the oldest ledger: %d and the latest ledger: %d for this rpc instance.", ledgerRange.FirstLedger.Sequence, ledgerRange.LastLedger.Sequence)
+	}
+
+	if req.Pagination != nil && req.Pagination.Limit > maxLimit {
+		return fmt.Errorf("limit must not exceed %d", maxLimit)
+	}
+
+	return nil
+}
+
+type transactionInfo struct {
```

Comment:
this struct should be public because it is referenced in other packages (namely test/get_transactions_test.go)

Reply:
@tamirms This struct is not referenced anywhere except the get_transactions.go file.

Reply:
https://github.com/stellar/soroban-rpc/blob/2d660841df55a3e5d5b569c07948ea66aad1481d/cmd/soroban-rpc/internal/test/get_transactions_test.go#L112-L114

here there is code outside the methods package which refers to fields defined in this struct